.github/workflows/build-plugin-archive.yml

```diff
@@ -98,6 +103,20 @@ jobs:
               echo "archive-name=${{ github.event.repository.name }}" >> $GITHUB_OUTPUT
           fi
 
+      - name: Set up plugin dir name
+        env:
+          PLUGIN_DIR_NAME: ${{ inputs.PLUGIN_DIR_NAME }}
+          ARCHIVE_NAME: ${{ inputs.ARCHIVE_NAME }}
```

Comment:
Is there a reason to define the environment variables on step level (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsenv) instead of on job level (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv; line 80)?
If you decide to move it to job level, please also move all other env declarations from step to job level. Thanks!

Reply:
It just feels better to define variables with the least possible scope. But I don't mind moving them if there is any reason for that.

Reply:
Moving to the job level saves us from setting it multiple times. So I would move it, yes.

Reply:
Ok, moved.

BTW, I noticed we also have sensitive variables declared at the job level. Isn't it a potential security issue? Malicious code in one of the used 3rd party actions could get access to them.